### DISCUSSION TITLE
How to format dates with recommended AP style a.m. and p.m.

### DISCUSSION THREAD

**Discussion by laurabennett** [craftcms/cms#13616] [Q&A] [answered]
https://github.com/craftcms/cms/issues/4558 mentions that en-GB formats with the dot style which is interesting but I don't want to change everything to "British". Is there a way to use the dot format? Looks like PHP only give uppercase or lowercase am or AM.

**Comment by brandonkelly** [comment, marked as answer]
The exact date/time formats come from the ICU library installed on the server, which is a massive database that tells apps like Craft how they should handle several aspects of internationalization, including how to format dates/times. So not something Craft should have a ton of control over. (And the locale data export mentioned in https://github.com/craftcms/cms/issues/4558#issuecomment-511256480 is no longer a thing in Craft 4, as the Intl extension (which provides an API for the ICU DB) is now a strict requirement.)

Probably the easiest way to force adding `.`s would be using the [`replace`](https://craftcms.com/docs/4.x/dev/filters.html#replace) filter:

```twig
{{ now|datetime|replace({
  AM: 'A.M.',
  PM: 'P.M.',
}) }}
```

You could throw that in a [macro](https://twig.symfony.com/doc/2.x/tags/macro.html) to keep your templates DRY:

```twig
{% macro datetime(date) %}
  {{ date|datetime|replace({
    AM: 'A.M.',
    PM: 'P.M.',
  }) }}
{% endmacro %}

...

{{ _self.datetime(now) }}
{{ _self.datetime(entry.postDate) }}
```
